### DISCUSSION TITLE
CASE in orderBy

### DISCUSSION THREAD

**Discussion by mjroeleveld** [mikro-orm/mikro-orm#4783] [Q&A]
```typescript
qb.orderBy([
        {
          [`CASE WHEN table.property = 'abc' THEN 0 ELSE 1 END`]:
            'DESC',
        }
]);
```

I would expect this works because of the [docs here](https://mikro-orm.io/docs/query-builder#using-custom-sql-fragments).

But it doesn't :). When I turn debugging on, the query simply doesn't contain the `ORDER_BY` clause. Any other way to have this work on 5.8?

**Comment by mjroeleveld** [comment]
Nevermind, it does work 🤦🏻‍♂️

**Comment by B4nan** [comment]
FYI you can also do this:

https://mikro-orm.io/docs/entity-manager#custom-property-ordering

**Comment by mjroeleveld** [comment]
Thanks! In this case I need to order across multiple properties so this doesn't work I think.
